src/components/dataconnector.vue

```diff
@@ -149,34 +128,69 @@ export default {
     },
     updateLayer: function (layerId) {
       this.selectedLayer = layerId
```

Comment:
The issue of int/string isn't here, but on the event handler `onLayerChange`. I updated the arcgis library to make the layers function return an object, with the `id` as the key.

Since the `id` of the layer is returned by ArcGIS's API, it will always start out being an int. I added a `parseInt` in the event handler.